share/ci/pypicongpu_generator.py

```diff
@@ -127,6 +129,50 @@ def get_all_major_pypi_versions(package_name):
     return [str(v) for v in version_map.values()]
 
 
+@typeguard.typechecked
+def combine_requirements(one: pkg_resources.Requirement, two: pkg_resources.Requirement) -> pkg_resources.Requirement:
+    if one.project_name != two.project_name:
+        print(f"requirements for {one.project_name} and {two.project_name} are incomparable!")
+
+    # accumulate all specs
+    specs: list = copy.deepcopy(one.specs)
+    specs.extend(two.specs)
+
+    last_tuple = specs.pop()
```

Comment:
We pop, but is it ever possible for the specs list to be empty?

Reply:
should not be possible, since if no requirements are found the list of parsed requirements we get will be an empty list, therefore skipping the entire loop body of `readInRequirementFiles`.

Nevertheless good catch hadn't thought about that.